src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfEhFrame.cs

```diff
@@ -122,14 +123,16 @@ private void WriteFde(in DwarfFde fde, uint cieOffset)
                     (fde.Cie.PersonalityEncoding != 0 ? AddressSize(fde.Cie.PersonalityEncoding) : 0) +
                     (fde.Cie.LsdaEncoding != 0 ? AddressSize(fde.Cie.LsdaEncoding) : 0) : 0;
 
+            uint pointerEncodingSize = AddressSize(fde.Cie.PointerEncoding);
+
             uint length =
                 4u + // Length
                 4u + // CIE offset
-                AddressSize(fde.Cie.PointerEncoding) + // PC start
-                AddressSize(fde.Cie.PointerEncoding) + // PC end
+                pointerEncodingSize + // PC start
+                pointerEncodingSize + // PC end
                 augmentationLength +
                 (uint)fde.Instructions.Length;
-            uint padding = ((length + 7u) & ~7u) - length;
+            uint padding = ((length + pointerEncodingSize - 1u) & ~(pointerEncodingSize - 1u)) - length;
```

Comment:
Consider extracting a helper for the padding calculation.

Reply:
Sounds reasonable. (I'll wait for the CI to avoid interrupting it and waiting for next re-run.)